lib/cambiatus/shop/category.ex

```diff
@@ -59,7 +61,79 @@ defmodule Cambiatus.Shop.Category do
     put_assoc(changeset, :categories, categories)
   end
 
-  def positional(query \\ Category) do
+  def validate_position(changeset) do
+    position = get_field(changeset, :position)
+
+    if position < 0 do
+      add_error(changeset, :position, "position cant be negative")
+    else
+      changeset
+    end
+  end
+
+  # Do nothing if there is a parent
+  def validate_root_position(%{changes: %{parent_id: _}} = changeset), do: changeset
+
+  # When inserting
+  def validate_root_position(
+        %{changes: %{community_id: community_id, position: position}} = changeset
+      ) do
+    count = Shop.count_categories(community_id)
+
+    unless position <= count + 1 do
+      add_error(
+        changeset,
+        :position,
+        "for new categories, position must be smaller or equal than #{count + 1}"
+      )
+    else
+      changeset
+    end
+  end
+
+  # When updating
+  def validate_root_position(
+        %{
+          data: %{id: id, community_id: community_id},
+          changes: %{position: position}
+        } = changeset
+      )
+      when not is_nil(id) do
+    count = Shop.count_categories(community_id)
+
+    unless position <= count do
+      add_error(
+        changeset,
+        :position,
+        "for existing categories, position must be smaller or equal than #{count}"
+      )
+    else
+      changeset
+    end
```

Comment:
Mesma coisa do `if`:

```suggestion
    if position > count do
      add_error(
        changeset,
        :position,
        "for existing categories, position must be smaller or equal than #{count}"
      )
    else
      changeset
    end
```